### ISSUE TITLE
line numbers

### ISSUE THREAD

**Issue by hueldoeu** [billthefarmer/editor#165]
please add line numbers so i can get rid of jota+

**Comment by billthefarmer** [comment]
I recently added to the [README](https://github.com/billthefarmer/editor#readme) for this app:

> There have been a number of issues raised on this app where users have obviously not read this README, looked at the [documentation](https://billthefarmer.github.io/editor/), or looked at old closed issues. Please read the README, read the [docs](https://billthefarmer.github.io/editor/), and look at the old issues before raising another one, you can search the issues from the box at the top.

This has been raised several times &ndash; #161, #97, #87. The comments I made previously still apply. Because this issue has been raised several times, I decided to test it. It works fine for small files that aren't long enough to scroll, but falls down badly with large files. There is a delay of several seconds updating the numbers for a ~2,000 line source file, which is a fraction of the size file the app can handle. To mitigate this, I tested just updating the numbers for the visible part of the file. This works fine, but because the scroll views used allow pixel by pixel scrolling in both directions, the line numbers don't line up. I think it is horrible.

![Screenshot_20211206-101329 1](https://user-images.githubusercontent.com/3448386/144828388-d04a6e16-3b00-4f9d-95c8-8452eecd3aac.png)
